apps/web/vibes/soul/sections/product-detail/index.tsx

```diff
@@ -24,21 +24,21 @@ interface Props<F extends Field> {
   product: Streamable<ProductDetailProduct | null>;
   action: ProductDetailFormAction<F>;
   fields: Streamable<F[]>;
-  ctaLabel?: string;
   quantityLabel?: string;
   incrementLabel?: string;
   decrementLabel?: string;
+  submitButton?: Streamable<{ label: string; disabled?: boolean }>;
```

Comment:
How come we're grouping in an object? We haven't been doing this for other labels. Also, shouldn't the submit button's disabled state be handled internally when the form is being submitted, etc.?

In what instance would we want to pass this value in as a hardcoded `false` value?

Reply:
Lets think about this. Purchase buttons can be disabled from the API if the product is not available, but the label of the button also depends on this status or if the order is preorderable. We want this to be streameable, since we fetch this state from the API. If this is the case, does having an object makes sense to you? Or should both props just be Streamable props?

Reply:
I split into two props as suggested. I think this makes sense.

Reply:
> When a product is out of stock.

That makes sense. Thank you!

re: streaming. I think it's ok to have separate props. Something I want to encourage more in VIBES is combining streamable values when it makes sense:

```tsx
// In a server component
<Streamable value={Promise.all([streamableFoo, streamableBar]}>

// In a client component
<Streamable value={[streamableFoo, streamableBar]}>
```

The server component example is one I often do. The client component one I have to figure out if it's possible while keeping a reasonable API.